todoman/cli.py

```diff
@@ -325,12 +337,22 @@ def list(ctx, lists, all, urgent, location, category, grep, sort, reverse):
                (not category or category in todo.categories) and
                (not pattern or (
                    pattern.search(todo.summary) or
-                   pattern.search(todo.description)
-                   ))
+                   pattern.search(todo.description))) and
+               (not due or
+                (todo.due and
+                 not todo.due.tzinfo and
+                 (todo.due <= max_naive)) or
+                (todo.due and
+                 todo.due.tzinfo and
+                 (todo.due <= max_aware)))
```

Comment:
Moving this into a function is really easy compared to implementing a new cache.